pages/docs/learn/inngest-steps.mdx

```diff
@@ -0,0 +1,164 @@
+    });
+  }
+);
+```
+
+The ID is also used as an identifier in the [Inngest Dev Server](/docs/local-development) for you to quickly find a specific step.
+
+## Available Step Methods
+
+### <a href="/docs/reference/functions/step-run" className="flex items-center gap-4"><IconConcurrency size="2rem"/><pre>step.run()</pre></a> {{anchor: false}}
+
+This method executes a defined piece of code. Use it to run synchronous or asynchronous code as a retriable step in your function.
+
+```typescript
+export default inngest.createFunction(
+  { id: "import-product-images" },
+  { event: "shop/product.imported" },
+  async ({ event, step }) => {
+    const uploadedImageURLs = await step.run("copy-images-to-s3", async () => {
+      return copyAllImagesToS3(event.data.imageURLs);
+    });
+  }
+);
+```
+
+### <a href="/docs/reference/functions/step-sleep" className="flex items-center gap-4"><IconConcurrency size="2rem"/><pre>step.sleep()</pre></a> {{anchor: false}}
+
+This method pauses execution for a specified duration. Even though it seems like a `setInterval`, your function does not run for that time. Inngest handles the scheduling for you. Use it to add delays or to wait for a specific amount of time before proceeding.
+
+```typescript
+export default inngest.createFunction(
+  { id: "send-delayed-email" },
+  { event: "app/user.signup" },
+  async ({ event, step }) => {
+    await step.sleep("wait-a-couple-of-days", "2d");
+    // Do something else
+  }
+);
+```
+
+### <a href="/docs/reference/functions/step-sleep-until" className="flex items-center gap-4"><IconConcurrency size="2rem"/><pre>step.sleepUntil()</pre></a> {{anchor: false}}
+
+This method pauses execution until a specific date time. Any date time string in the format accepted by the Date object, for example `YYYY-MM-DD` or `YYYY-MM-DDHH:mm:ss`.
+
+```typescript
+export default inngest.createFunction(
+  { id: "send-scheduled-reminder" },
+  { event: "app/reminder.scheduled" },
+  async ({ event, step }) => {
+    const date = new Date(event.data.remind_at);
+    await step.sleepUntil("wait-for-the-date", date);
+    // Do something else
+  }
+);
+```
+
+### <a href="/docs/reference/functions/step-wait-for-event" className="flex items-center gap-4"><IconConcurrency size="2rem"/><pre>step.waitForEvent()</pre></a> {{anchor: false}}
+
+This method pauses the execution until a specific event is received.
+
+```typescript
+export default inngest.createFunction(
+  { id: "send-onboarding-nudge-email" },
+  { event: "app/account.created" },
+  async ({ event, step }) => {
+    const onboardingCompleted = await step.waitForEvent(
+      "wait-for-onboarding-completion",
+      { event: "app/onboarding.completed", timeout: "3d", match: "data.userId" }
+    );
+    // Do something else
+  }
+);
+```
+
+### <a href="/docs/reference/functions/step-send-event" className="flex items-center gap-4"><IconConcurrency size="2rem"/><pre>step.sendEvent()</pre></a> {{anchor: false}}
```

Comment:
I might order invoke above sendEvent as I think it's usually what developers want. It then could be explained as "Use send event when you want to trigger other functions, but you do not need to return the result." I think the order of invoke (you need the result) feels more like a good default then the option of sending an event (don't need the result) feels like a logical progression.